timelock-agent/src/main/java/com/palantir/timelock/config/PaxosInstallConfiguration.java

```diff
@@ -39,6 +44,9 @@ default File dataDirectory() {
         return new File("var/data/paxos");
     }
 
+    @JsonProperty("sqlite-persistence")
+    Optional<SqlitePaxosPersistenceConfiguration> sqlitePersistence();
```

Comment:
What if this was required and you actually checked state (meaning whether you already migrated or not/running validation) not just by looking if directories exist, but actually checking an internal sqlite table?

Reply:
The purpose of `isNewService` is not to determine whether migration/validation has taken place - it's to avoid accidental misconfiguration of the directories. 

That said an inclusive OR is probably not ideal. While we're running in validation mode it's probably best to just test the base directory. Once we are migrating:

- If both directories exist, accept.
- If the data directory exists and SQLite does not, accept: migration maybe hasn't taken place. The SQLite migration will poison the data directory so we will be safe overall
- Otherwise, reject: we create the data directory every time when is new service _is_ true (*).

Long term solution / if we don't want to keep (*) around:
- If both directories are absent, reject.
- If SQLite exists and data directory does not, check if SQLite has migrated, and accept if yes, reject if no.

We could make this required, but I would not want to run the validation/migration everywhere immediately (because of its possible effect on startup times).

Reply:
That makes sense. I would challenge if we actually should run this everywhere straight away: our infra should just downgrade if we overrun the startup budget, which means we'll need to fix that anyway

Reply:
Cool. I was thinking of using the canaries as a mechanism for avoiding a possible rollback at large deployment (I'd assume the migration is O(N log N) with number of entries) but maybe that's too much effort, and I should just do the math here when we do pub lish the release and abort if we believe things will explode (e.g. if Herb takes 5 minutes, Brimstone will definitely be a no-go)

Reply:
yeah, if we can get away without special config I'd say it's worth a try. Worst thing that happens as I understand it is that you just timeout the upgrade, and infra should rollback